packages/babel-plugin-i18n-calypso/src/index.js

```diff
@@ -81,13 +106,43 @@ function getNodeAsString( node ) {
 	}
 }
 
+/**
+ * Returns true if the specified key of a function is valid for assignment in
+ * the translation object.
+ *
+ * @param {string} key Key to test.
+ *
+ * @returns {boolean} Whether key is valid for assignment.
+ */
+function isValidTranslationKey( key ) {
+	return -1 !== VALID_TRANSLATION_KEYS.indexOf( key );
+}
+
 module.exports = function() {
 	let strings = {},
 		nplurals = 2,
-		baseData;
+		baseData,
+		hasAliasedTranslate = false;
 
 	return {
 		visitor: {
+			ImportDeclaration( path ) {
+				// Set `hasAliasedTranslate` flag if `translate` from
+				// `i18n-calypso`` is aliased as `__`.
+				if (
+					'i18n-calypso' === path.node.source.value &&
+					path.node.specifiers.some(
+						specifier =>
+							specifier.imported &&
+							specifier.local &&
+							'translate' === specifier.imported.name &&
+							'__' === specifier.local.name
+					)
+				) {
+					hasAliasedTranslate = true;
```

Comment:
Thinking about it, could we do without the `hasAliasedTranslate` and just change the `DEFAULT_FUNCTIONS_ARGUMENTS_ORDER['__']` declaration?

Reply:
Removed it in https://github.com/Automattic/wp-calypso/pull/39280/commits/708e88a482b523f4e7c9373231a5b9d1774d07d9